nixos/modules/installer/tools/nixos-generate-config.pl

```diff
@@ -605,35 +659,143 @@ sub multiLineList {
+  #
+  # services.xserver.videoDrivers = [ "<name-of-the-driver>" ];
+
+  # Enable support for 32-bit OpenGL libraries on x86_64 which is
+  # useful if you want to run 32-bit OpenGL applications
+  #
+  # hardware.opengl.driSupport32Bit = true;
+  #
+  # E.g., games under Wine
+  #
+  # environment.systemPackages = with pkgs; [ wine winetricks ];
+
+  # Some of the X11 drivers above might not work with the current Linux
+  # kernel, so you might end up overriding that too.
+  #
+  # boot.kernelPackages = pkgs.linuxPackages_4_9;
+
+  # Define a user account. Don't forget to set a password with ‘passwd’.
+  #
+  # users.users.me = {
+  #   isNormalUser = true;
+  #   uid = 1000;
+  #   extraGroups = [ "input" "audio" "video" "cdrom" ]
+  #              ++ [ "wheel" ]; # for sudo
+  #   openssh.authorizedKeys.keyFiles = [ /etc/nixos/id_rsa.pub ];
+  # };
+
   # Enable the KDE Desktop Environment.
+  #
   # services.xserver.displayManager.sddm.enable = true;
   # services.xserver.desktopManager.plasma5.enable = true;
 
-  # Define a user account. Don't forget to set a password with ‘passwd’.
-  # users.users.guest = {
+  # Or maybe you want to enable Xfce Desktop Environment and Slim
+  # display manager?
+  #
+  # services.xserver.displayManager.slim.enable = true;
+  # services.xserver.desktopManager.xfce.enable = true;
+
+  # Or maybe you just want to automatically login into your .xsession
+  # script? Then don't enable any desktopManagers and uncomment the
+  # following.
+  #
+  # services.xserver.displayManager.auto = {
+  #   enable = true;
+  #   user = "me";
+  # };
+
+  # If you're in a hurry, you may stop editing your configuration.nix now
+  # and completely ignore everything below.
+
+  # It's generally considered a good practice to run networking
+  # software at the very least in a separate user accounts (if not in a
+  # separate containers or a VMs). NixOS does this automatically for
+  # the system services. However, like most other distros, at the moment
+  # NixOS doesn't do anything about the networking software that you run
+  # from your normal user accounts. Which is why you might want to setup
+  # separate user accounts for such things yourself.
+  #
+  # users.users.network = {
   #   isNormalUser = true;
-  #   uid = 1000;
+  #   uid = 1001;
+  #
+  #   # Like systemPackages, but for this user only.
+  #   packages = pkgs; [ firefox thunderbird qtox ];
+  #
+  #   # You probably want it to play sound via ALSA
+  #   extraGroups = [ "audio" ];
+  #   # but you probably don't want to give it direct DRI access
+  #   # since OpenGL video drivers are pretty buggy and giving DRI
+  #   # access to WebGL might not be the best of ideas.
+  #   #          ++ [ "video" ];
   # };
+  #
+  # The simplest way to get to the software installed there is by
+  # running
+  #
+  # \$ xhost +si:localuser:network
+  # \$ sudo -Hiu network firefox &disown
+  # \$ sudo -Hiu network qtox &disown
+  #
+  # from the account logged in into X11.
```

Comment:
Honestly this feels like it should be a whole new option,which will add the required wrappers and sudo configurations.
Also for desktop users you should probably point to `programs.firejail`